src/client/Client.cc

```diff
@@ -872,7 +872,11 @@ Inode * Client::add_update_inode(InodeStat *st, utime_t from,
 
   if (new_version ||
       (new_issued & (CEPH_CAP_ANY_FILE_RD | CEPH_CAP_ANY_FILE_WR))) {
-    in->layout = st->layout;
+    if(st->layout.is_valid()){
+      in->layout = st->layout;
+    } else {
+      ldout(cct, 10) << " bad layout " << dendl;
```

Comment:
and leave it unmodified??

Reply:
It is best to judge whether the layout is valid.